### PR TITLE
feat(Bonus Pagamenti Digitali): [#176575120] Graphical refactor entry point co-badge

### PR DESCRIPTION
**Pull request by debiff**
## Short description
This PR align the bancomat detail screen with the design.

## List of changes proposed in this pull request
- Refactor `BancomatInformation` component
- Added `BancomatInformationBottomSheet` component
- Insert `BancomatInformationBottomSheet` in `NewMethodAddedNotifier` component

https://user-images.githubusercontent.com/11773070/106922018-e25dd200-670c-11eb-868c-e1652a3c3680.mov

### CODE REVIEW

File: locales/en/index.yml [hunk 1]
```diff
@@ -864,6 +864,19 @@ wallet:
           title: "TMP KO Timeout add"
       add:
         title: "TMP Add title"
+      chooseType:
+        title: "Does your card allow you to shop online?"
+        description: "This question allows us to better understand what your card is like and help you add it to your Wallet."
```
Comment (thisisjp) [inline locales/en/index.yml:869]:
```suggestion
        description: "This allows us to better identify your card and help you add it to your Wallet."
```

File: locales/en/index.yml [hunk 2]
```diff
@@ -864,6 +864,19 @@ wallet:
...
+        path:
+          enabled:
+            title: "Yes, it is enabled for online purchases"
+            description: "The data will be verified in the way appointed by your bank, for example via text message or app."
+          disabled:
+            title: "No, I only use it in shops"
+            description: "We will look for all the cards in your name at the banks participating in the service."
+          unknown:
+            title: "I do not know"
```
Comment (thisisjp) [inline locales/en/index.yml:878]:
```suggestion
            title: "I don't know"
```

File: locales/en/index.yml [hunk 3]
```diff
@@ -864,6 +864,19 @@ wallet:
...
         title: "TMP Add title"
+      chooseType:
+        title: "Does your card allow you to shop online?"
+        description: "This question allows us to better understand what your card is like and help you add it to your Wallet."
+        path:
+          enabled:
+            title: "Yes, it is enabled for online purchases"
+            description: "The data will be verified in the way appointed by your bank, for example via text message or app."
+          disabled:
+            title: "No, I only use it in shops"
```
Comment (thisisjp) [inline locales/en/index.yml:875]:
```suggestion
            title: "No, I only use it for in-store purchases"
```

File: locales/en/index.yml [hunk 4]
```diff
@@ -864,6 +864,19 @@ wallet:
           title: "TMP KO Timeout add"
       add:
         title: "TMP Add title"
+      chooseType:
+        title: "Does your card allow you to shop online?"
+        description: "This question allows us to better understand what your card is like and help you add it to your Wallet."
+        path:
+          enabled:
+            title: "Yes, it is enabled for online purchases"
```
Comment (thisisjp) [inline locales/en/index.yml:872]:
```suggestion
            title: "Yes, it works online"
```

File: locales/en/index.yml [hunk 5]
```diff
@@ -864,6 +864,19 @@ wallet:
...
+            description: "The data will be verified in the way appointed by your bank, for example via text message or app."
```
Comment (thisisjp) [inline locales/en/index.yml:873]:
```suggestion
            description: "We'll verify card data with your bank, usually with an SMS or a push notification"
```

File: locales/it/index.yml [hunk 6]
```diff
@@ -884,6 +884,19 @@ wallet:
           title: "TMP KO Timeout add"
       add:
         title: "TMP Add title"
+      chooseType:
+        title: "La tua carta ti permette di acquistare online?"
+        description: "Questa domanda ci permette di capire meglio com’è fatta la tua carta e aiutarti ad aggiungerla al tuo Portafoglio."
+        path:
+          enabled:
+            title: "Sì, è abilitata agli acquisti online"
+            description: "I dati verranno verificati con le modalità previste dalla tua banca, ad esempio via sms o app."
```
Comment (thisisjp) [inline locales/it/index.yml:893]:
```suggestion
            description: "Verificheremo i dati della carta con la tua banca, di solito via sms o notifica push"
```

File: locales/en/index.yml [hunk 7]
```diff
@@ -864,6 +864,19 @@ wallet:
...
+          disabled:
+            title: "No, I only use it in shops"
+            description: "We will look for all the cards in your name at the banks participating in the service."
```
Comment (thisisjp) [inline locales/en/index.yml:876]:
```suggestion
            description: "We will search for all cards registered to you at participating banks"
```

File: locales/en/index.yml [hunk 8]
```diff
@@ -864,6 +864,19 @@ wallet:
...
+          unknown:
+            title: "I do not know"
+            description: "We will look for all the cards in your name at the banks participating in the service."
```
Comment (thisisjp) [inline locales/en/index.yml:879]:
```suggestion
            description: "We will search for all cards registered to you at participating banks"
```

File: locales/en/index.yml [hunk 9]
```diff
@@ -1178,9 +1191,9 @@ wallet:
         two: "can't be used to make in-app payments"
         three: ", but they can only be used at physical stores (POS)."
       debit:
-        title: "Does your card have one of these logos?"
-        body: "If your PagoBANCOMAT card has other international circuits, you can also add it as a **debit or credit card** by entering its number on the front. When you activate Cashback on this method, you will also be able to collect transactions on these circuits."
-        addCta: "Add as Debit/Credit Card"
+        title: "Does your card have more than one logo?"
+        body: "If your PagoBANCOMAT card shows other international circuits, you can add them by searching for all debit (or credit) cards in your name. By activating the Cashback method, you can also accumulate transactions made on these circuits."
```
Comment (thisisjp) [inline locales/en/index.yml:1195]:
```suggestion
        body: "If your PagoBANCOMAT card shows other international circuits, you can add them by searching for all debit (or credit) cards registered to you. When you enable Cashback on this method, you will also be able to collect transactions on these circuits."
```

File: locales/it/index.yml [hunk 10]
```diff
@@ -884,6 +884,19 @@ wallet:
...
+          disabled:
+            title: "No, la uso solo nei negozi"
+            description: "Cercheremo tutte le carte intestate a te presso le banche aderenti al servizio."
```
Comment (thisisjp) [inline locales/it/index.yml:896]:
```suggestion
            description: "Cercheremo tutte le carte intestate a te presso le banche aderenti al servizio"
```

File: locales/it/index.yml [hunk 11]
```diff
@@ -884,6 +884,19 @@ wallet:
...
+          unknown:
+            title: "Non lo so"
+            description: "Cercheremo tutte le carte intestate a te presso le banche aderenti al servizio."
```
Comment (thisisjp) [inline locales/it/index.yml:899]:
```suggestion
            description: "Cercheremo tutte le carte intestate a te presso le banche aderenti al servizio"
```

**Review by thisisjp — CHANGES_REQUESTED** [review]
I made some small changes on copy

File: ts/features/wallet/bancomat/screen/BancomatInformation.tsx [hunk 12]
```diff
@@ -26,50 +27,48 @@ const styles = StyleSheet.create({
...
-      bordered={true}
-      onPress={() => {
-        props.onAddPaymentMethod?.();
-      }}
-      onPressWithGestureHandler={true}
-    >
-      <Label>{I18n.t("wallet.bancomat.details.debit.addCta")}</Label>
-    </ButtonDefaultOpacity>
-  </View>
-);
```
Comment (fabriziofff) [inline ts/features/wallet/bancomat/screen/BancomatInformation.tsx:72]: In my understanding `hideCobrandTitle` is not used anymore. If this is the case, how about removing it from the props?

File: ts/features/wallet/component/NewMethodAddedNotifier.tsx [hunk 13]
```diff
@@ -51,7 +24,9 @@ const NewPaymentMethodAddedNotifier = (props: Props) => {
     string
   >("");
 
-  const { present } = newBancomatBottomSheet(props.startCoBadgeOnboarding);
+  const { present } = BancomatInformationBottomSheet(
```
Comment (fabriziofff) [inline ts/features/wallet/component/NewMethodAddedNotifier.tsx:27]:
```suggestion
  const { present } = bancomatInformationBottomSheet(
```
To keep consistency in the codebase, we use `PascalCase` for components and types, `camelCase` for functions and const.

File: ts/features/wallet/bancomat/screen/BancomatInformation.tsx [hunk 14]
```diff
@@ -26,50 +27,48 @@ const styles = StyleSheet.create({
...
+const BancomatInformation: React.FunctionComponent<Props> = props => {
+  const { present } = BancomatInformationBottomSheet(props.onAddPaymentMethod);
```
Comment (fabriziofff) [inline ts/features/wallet/bancomat/screen/BancomatInformation.tsx:44]:
```suggestion
  const { present } = bancomatInformationBottomSheet(props.onAddPaymentMethod);
```

To keep consistency in the codebase, we use `PascalCase` for components and types, `camelCase` for functions and const.

**Review by fabriziofff — CHANGES_REQUESTED** [review]

**Review by fabriziofff — APPROVED** [review]

**Pull request merged**